maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -944,6 +944,7 @@ public void execute()
                         return;
                 }
             }
+            verifyParameters( true );
```

Comment:
second call? in one methd.

Reply:
@CMoH @slawekjaranowski 
Here I can see that the method `verifyParameters()` is called twice. Why?

Reply:
The short answer is: 
- the first call is meant to run the fast parameters check, also determining if the plugin is enabled
- the second is meant to run more checks, those that are only applicable to modules that do have test classes. 

I believe such a split is necessary, since the test `getTestClassesDirectory().exists()` at the beginning of `verifyParameters()` is insufficient. The test directory might exist, empty, or containing files/classes which don't provide tests to run.

The actual problem for me here is `warnIfDefunctGroupsCombinations()`, which should not crash for modules that don't have tests due to parameters inherited from a parent pom.


The correct, more resource-expensive check, is done by `scanForTestClasses()`, and the second call to `verifyParameters()` only has any effect if there are indeed tests to run (and `failIfNoTests==false`).


The second benefit is that the splitting the responsibilities of `verifyParameters()` into two methods (although I only started with a quick fix via a boolean parameter) would help as placeholders to continue refining the validation code.

I touched on this also in [this comment](https://github.com/apache/maven-surefire/pull/388#discussion_r748769750) below.

I am able to assist with changes, if you feel this path is helpful to the plugin.